### PR TITLE
Fix: Exception if init epoch is not yet passed

### PR DESCRIPTION
**Pull request by F4ever**

### CODE REVIEW

File: src/modules/ejector/ejector.py [hunk 1]
```diff
@@ -306,10 +306,11 @@ def _get_churn_limit(self, blockstamp: ReferenceBlockStamp) -> int:
...
     def _get_total_active_validators(self, blockstamp: ReferenceBlockStamp) -> int:
-        total_active_validators = len([
-            is_active_validator(val, blockstamp.ref_epoch)
-            for val in self.w3.cc.get_validators(blockstamp)
-        ])
+        total_active_validators = reduce(
+            lambda total, validator: total + int(is_active_validator(validator, blockstamp.ref_epoch)),
+            self.w3.cc.get_validators(blockstamp),
+            0,
+        )
```
Comment (madlabman) [inline src/modules/ejector/ejector.py:313]:
```suggestion
        total_active_validators = sum(
            1 for v in self.w3.cc.get_validators(blockstamp) if is_active_validator(v, blockstamp.ref_epoch)
        )
```

File: tests/modules/submodules/consensus/test_consensus.py [hunk 2]
```diff
@@ -130,6 +132,32 @@ def test_get_blockstamp_for_report_slot_not_finalized(web3, consensus, caplog, s
...
+    consensus._get_consensus_contract = Mock(return_value=consensus_contract)
+    consensus.get_frame_config = Mock(return_value=FrameConfigFactory.build(initial_epoch=5, epochs_per_frame=10))
+    consensus.get_chain_config = Mock(return_value=ChainConfigFactory.build())
+
+    first_frame = consensus.get_current_frame(bs)
+
+    assert first_frame.ref_slot == 5 * 32
+    assert first_frame.report_processing_deadline_slot == (5 + 10) * 32 - 1
+
+
```
Comment (madlabman) [inline tests/modules/submodules/consensus/test_consensus.py:160]: I would expect `FAR_FUTURE_EPOCH` to be handled separate.

```suggestion
@pytest.mark.unit
def test_initial_epoch_is_set_to_far_future_epoch(web3, consensus, caplog, set_no_account):
    bs = ReferenceBlockStampFactory.build()

    consensus_contract = Mock(get_current_frame=Mock(side_effect=ContractCustomError('0xcd0883ea', '0xcd0883ea')))
    consensus._get_consensus_contract = Mock(return_value=consensus_contract)
    consensus.get_frame_config = Mock(
        return_value=FrameConfigFactory.build(
            initial_epoch=48038396025285290,
            epochs_per_frame=10,
        )
    )
    consensus.get_chain_config = Mock(return_value=ChainConfigFactory.build())

    with pytest.raises(ValueError):
        consensus.get_current_frame(bs)
```

**Review by vgorkavenko — APPROVED** [review]

File: src/modules/submodules/consensus.py [hunk 3]
```diff
@@ -90,9 +95,23 @@ def get_chain_config(self, blockstamp: BlockStamp) -> ChainConfig:
         return consensus_contract.get_chain_config(blockstamp.block_hash)
 
     @lru_cache(maxsize=1)
-    def get_current_frame(self, blockstamp: BlockStamp) -> CurrentFrame:
+    def get_initial_or_current_frame(self, blockstamp: BlockStamp) -> CurrentFrame | None:
```
Comment (madlabman) [inline src/modules/submodules/consensus.py:98]: No `None`.

**Review by madlabman — APPROVED** [review]

File: src/modules/submodules/consensus.py (hunk 3 above)
Comment (F4ever) [reply src/modules/submodules/consensus.py]: fixed

**Review by vgorkavenko — APPROVED** [review]

**Review by madlabman — APPROVED** [review]

**Pull request merged**